src/main/java/dev/binclub/javaception/classfile/constants/ClassInfo.java

```diff
@@ -14,6 +19,13 @@ public Object resolve(Object[] cp) {
 		return this;
 	}
 	
+	public Klass getKlass() {
+		if (this.klass == null) {
+			this.klass = SystemDictionary.findReferencedClass(null, new ClassType(name));
```

Comment:
This should be making use of the `referencedBy`. Hopefully by the time this constant is resolved the class stream that had this constant has been resolved into a `Klass`

Reply:
this can be done in the future